bin/circle-dependencies-post

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+set -e
+
+# We need jq to analyse the PR files JSON, so install that first
+sudo apt-get -q update && sudo apt-get -q install bc jq
+
+rm -f "$STATE" && touch "$STATE"
+
+if [ -n "$CI_PULL_REQUEST" -a "$CI_PULL_REQUEST" = "$CI_PULL_REQUESTS" ] ; then
+    if PR=$(echo "$CI_PULL_REQUEST" | grep -oP '(?<=^https://github.com/weaveworks/weave/pull/)[0-9]+$') ; then
+        echo "GH_PR=$PR" >> "$STATE"
+        if FILES=$(curl -s https://api.github.com/repos/weaveworks/weave/pulls/$PR/files | jq -r '.[]|.filename') ; then
+            echo "PR touches [$FILES]"
+            if ! echo "$FILES" | grep -v ^site/ ; then
+                echo "PR contains site changes only - skipping tests"
+                echo "SKIP_TESTS=1" >> "$STATE"
```

Comment:
I think this variable would be better inverted and named as something like `RUN_TESTS_AND_PUBLISH`, since it is later used to drive decisions like "should I publish images?".  Or maybe a higher-level fact like `MAY_CHANGE_BEHAVIOUR`